sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -154,12 +158,12 @@ abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
       // --- BroadcastHashJoin --------------------------------------------------------------------
 
       case ExtractEquiJoinKeys(joinType, leftKeys, rightKeys, condition, left, right)
-        if canBuildRight(joinType) && canBroadcast(right) =>
+        if canBuildRight(joinType) && canBroadcast(right, left.stats.hints.broadcast) =>
```

Comment:
What about just adding here `&& !left.stats.hints.broadcast`? In this way we are not changing the `canBroadcast` method which is not really readable after the change.

Reply:
@mgaido91 Thanks for your suggestion.